src/Missile.cpp

```diff
@@ -172,24 +172,51 @@ bool Missile::OnDamage(Object *attacker, float kgDamage, const CollisionContact
 	return true;
 }
 
+double calcAreaSphere(const double r) {
+	return 4.0*M_PI*r*r;
+}
+
+double calcAreaCircle(const double r) {
+	return M_PI*r*r;
+}
+
 void Missile::Explode()
 {
 	Pi::game->GetSpace()->KillBody(this);
 
-	const double damageRadius = 200.0;
-	const double kgDamage = 10000.0;
+	double kgYield = 200.0; // yield of the used explosive in kg
+	const double jEnergyPerKgExplosive = 4.184e6; // energy density of explosive, here it's TNT
+
+	const double jYield = kgYield * jEnergyPerKgExplosive; // how much energy was converted in the explosion?
 
+	double queryRadius = 2000.0; // hardcoded to 2 km, this is sufficient for most explosions
```

Comment:
Why such a large distance compared to the old one?

Reply:
This is just the query distance, not the "real" explosion distance. There is no real explosion distance due to the nature of an explosion in empty space. The spatial data structure which is queried has to know a maximum query distance.
I've chosen a relatively large distance to allow for a bigger yield without modifying the queryDistance too.
Note here that this doesn't have any effect on the performance because the query is done once and because space is relatively empty anyways. Not an issue in my opinion here.

Reply:
May I suggest adding a comment to that effect?

```suggestion
	// The radius in which to search for possible recipients of the damages
	// Hardcoded to 2 km, this should be sufficient for most explosions
	const double queryRadius = 2000.0;
```